front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,73 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (!theoreticalMarginIndex) {
+    return ['', '', ''];
```

Comment:
I think it would be better to return an object like this, what do you think ?
```suggestion
    return {};
```

Reply:
yes, changed from the other comment on the other return